File: pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.pojogroup.Cart;

import java.util.List;

//操作购物车接口
public interface CartService {
    //添加购物车
    public List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num);

    //从redis中查询购物车
    public List<Cart> findCartListFromRedis(String userName);

    //将购物车存入redis中
    public void saveCartListToRedis(String userName, List<Cart> cartList);

    //合并购物车
    public List<Cart> mergeCartList(List<Cart> cartList1, List<Cart> cartList2);
}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
app.service("CartService", function ($http) {
    this.findCartList = function () {
        $http.get("../cart/findCartList.do");
    }
    this.addGoodsToCartList = function (itemId, num) {
        $http.get("../cart/findCartList.do?itemId=" + itemId + "&num=" + num);
    }

    //求合计
    this.sum = function (cartList) {
        var totalValue = {totalNum: 0, totalMoney: 0.00};
        for (var i = 0; i < cartList.length; i++) {
            var cart = cartList[i];
            for (var j = 0; j < cart.orderItemList.length; j++) { //购物车明细
                totalValue.totalNum += cart.orderItemList[i].num;  //商品明细数量
                totalValue.totalMoney += orderItem.totalFee;  //商品明细总价
            }
        }
        return totalValue;
    }

})